framework_tvb/tvb/adapters/simulator/monitor_forms.py

```diff
@@ -84,6 +85,23 @@ def get_form_for_monitor(monitor_class):
     return get_monitor_to_form_dict().get(monitor_class)
 
 
+def build_list_of_monitors_from_names(monitor_names, is_surface_simulation):
+    next_monitors_dict = dict()
+    monitor_dict = get_ui_name_to_monitor_dict(is_surface_simulation)
+
+    count = 0
+    for monitor_name in monitor_names:
+        monitor = monitor_dict[monitor_name]
+        next_monitors_dict[monitor.__name__] = (monitor(), count + 1)
+        count = count + 1
```

Comment:
I see this mechanism in 2 places, could it be extracted to a common method?

Reply:
I don't know, I think it would be a pretty forced extraction because one of them iterates over names and builds the list (when configuring a simulation) and the other one from view models (when loading a simulation). Also, in the first one I mentioned get_ui_name_to_monitor_dict is also called and the code inside the for loop is not really the same, except for the fact that we have a counter variable that keeps increasing.